
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Singular Genomics G4X documentation">
      
      
        <meta name="author" content="Singular Genomics">
      
      
        <link rel="canonical" href="https://docs.singulargenomics.com/g4x_data/g4x_output/">
      
      
        <link rel="prev" href="../">
      
      
        <link rel="next" href="../data_import/">
      
      
      <link rel="icon" href="../../_core/assets/favicon.ico">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.20">
    
    
      
        <title>G4X-output - Singular Genomics Docs</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.e53b48f4.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      
  
  
    
      
      
    
  
    
      
      
    
  
    
      
      
    
  
    
      
      
    
  
    
      
      
    
  
    
      
      
    
  
    
      
      
    
  
    
      
      
    
  
    
      
      
    
  
    
      
      
    
  
    
      
      
    
  
  
  <style>:root{--md-admonition-icon--note:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%2016%2016%22%3E%3Cpath%20d%3D%22M1%207.775V2.75C1%201.784%201.784%201%202.75%201h5.025c.464%200%20.91.184%201.238.513l6.25%206.25a1.75%201.75%200%200%201%200%202.474l-5.026%205.026a1.75%201.75%200%200%201-2.474%200l-6.25-6.25A1.75%201.75%200%200%201%201%207.775m1.5%200c0%20.066.026.13.073.177l6.25%206.25a.25.25%200%200%200%20.354%200l5.025-5.025a.25.25%200%200%200%200-.354l-6.25-6.25a.25.25%200%200%200-.177-.073H2.75a.25.25%200%200%200-.25.25ZM6%205a1%201%200%201%201%200%202%201%201%200%200%201%200-2%22/%3E%3C/svg%3E');--md-admonition-icon--abstract:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%2016%2016%22%3E%3Cpath%20d%3D%22M2.5%201.75v11.5c0%20.138.112.25.25.25h3.17a.75.75%200%200%201%200%201.5H2.75A1.75%201.75%200%200%201%201%2013.25V1.75C1%20.784%201.784%200%202.75%200h8.5C12.216%200%2013%20.784%2013%201.75v7.736a.75.75%200%200%201-1.5%200V1.75a.25.25%200%200%200-.25-.25h-8.5a.25.25%200%200%200-.25.25m13.274%209.537zl-4.557%204.45a.75.75%200%200%201-1.055-.008l-1.943-1.95a.75.75%200%200%201%201.062-1.058l1.419%201.425%204.026-3.932a.75.75%200%201%201%201.048%201.074M4.75%204h4.5a.75.75%200%200%201%200%201.5h-4.5a.75.75%200%200%201%200-1.5M4%207.75A.75.75%200%200%201%204.75%207h2a.75.75%200%200%201%200%201.5h-2A.75.75%200%200%201%204%207.75%22/%3E%3C/svg%3E');--md-admonition-icon--info:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%2016%2016%22%3E%3Cpath%20d%3D%22M0%208a8%208%200%201%201%2016%200A8%208%200%200%201%200%208m8-6.5a6.5%206.5%200%201%200%200%2013%206.5%206.5%200%200%200%200-13M6.5%207.75A.75.75%200%200%201%207.25%207h1a.75.75%200%200%201%20.75.75v2.75h.25a.75.75%200%200%201%200%201.5h-2a.75.75%200%200%201%200-1.5h.25v-2h-.25a.75.75%200%200%201-.75-.75M8%206a1%201%200%201%201%200-2%201%201%200%200%201%200%202%22/%3E%3C/svg%3E');--md-admonition-icon--success:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%2016%2016%22%3E%3Cpath%20d%3D%22M13.78%204.22a.75.75%200%200%201%200%201.06l-7.25%207.25a.75.75%200%200%201-1.06%200L2.22%209.28a.75.75%200%200%201%20.018-1.042.75.75%200%200%201%201.042-.018L6%2010.94l6.72-6.72a.75.75%200%200%201%201.06%200%22/%3E%3C/svg%3E');--md-admonition-icon--question:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%2016%2016%22%3E%3Cpath%20d%3D%22M0%208a8%208%200%201%201%2016%200A8%208%200%200%201%200%208m8-6.5a6.5%206.5%200%201%200%200%2013%206.5%206.5%200%200%200%200-13M6.92%206.085h.001a.749.749%200%201%201-1.342-.67c.169-.339.436-.701.849-.977C6.845%204.16%207.369%204%208%204a2.76%202.76%200%200%201%201.637.525c.503.377.863.965.863%201.725%200%20.448-.115.83-.329%201.15-.205.307-.47.513-.692.662-.109.072-.22.138-.313.195l-.006.004a6%206%200%200%200-.26.16%201%201%200%200%200-.276.245.75.75%200%200%201-1.248-.832c.184-.264.42-.489.692-.661q.154-.1.313-.195l.007-.004c.1-.061.182-.11.258-.161a1%201%200%200%200%20.277-.245C8.96%206.514%209%206.427%209%206.25a.61.61%200%200%200-.262-.525A1.27%201.27%200%200%200%208%205.5c-.369%200-.595.09-.74.187a1%201%200%200%200-.34.398M9%2011a1%201%200%201%201-2%200%201%201%200%200%201%202%200%22/%3E%3C/svg%3E');--md-admonition-icon--warning:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%2016%2016%22%3E%3Cpath%20d%3D%22M6.457%201.047c.659-1.234%202.427-1.234%203.086%200l6.082%2011.378A1.75%201.75%200%200%201%2014.082%2015H1.918a1.75%201.75%200%200%201-1.543-2.575Zm1.763.707a.25.25%200%200%200-.44%200L1.698%2013.132a.25.25%200%200%200%20.22.368h12.164a.25.25%200%200%200%20.22-.368Zm.53%203.996v2.5a.75.75%200%200%201-1.5%200v-2.5a.75.75%200%200%201%201.5%200M9%2011a1%201%200%201%201-2%200%201%201%200%200%201%202%200%22/%3E%3C/svg%3E');--md-admonition-icon--failure:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%2016%2016%22%3E%3Cpath%20d%3D%22M2.344%202.343za8%208%200%200%201%2011.314%2011.314A8.002%208.002%200%200%201%20.234%2010.089a8%208%200%200%201%202.11-7.746m1.06%2010.253a6.5%206.5%200%201%200%209.108-9.275%206.5%206.5%200%200%200-9.108%209.275M6.03%204.97%208%206.94l1.97-1.97a.749.749%200%200%201%201.275.326.75.75%200%200%201-.215.734L9.06%208l1.97%201.97a.749.749%200%200%201-.326%201.275.75.75%200%200%201-.734-.215L8%209.06l-1.97%201.97a.749.749%200%200%201-1.275-.326.75.75%200%200%201%20.215-.734L6.94%208%204.97%206.03a.75.75%200%200%201%20.018-1.042.75.75%200%200%201%201.042-.018%22/%3E%3C/svg%3E');--md-admonition-icon--danger:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%2016%2016%22%3E%3Cpath%20d%3D%22M9.504.43a1.516%201.516%200%200%201%202.437%201.713L10.415%205.5h2.123c1.57%200%202.346%201.909%201.22%203.004l-7.34%207.142a1.25%201.25%200%200%201-.871.354h-.302a1.25%201.25%200%200%201-1.157-1.723L5.633%2010.5H3.462c-1.57%200-2.346-1.909-1.22-3.004zm1.047%201.074L3.286%208.571A.25.25%200%200%200%203.462%209H6.75a.75.75%200%200%201%20.694%201.034l-1.713%204.188%206.982-6.793A.25.25%200%200%200%2012.538%207H9.25a.75.75%200%200%201-.683-1.06l2.008-4.418.003-.006-.004-.009-.006-.006-.008-.001q-.005%200-.009.004%22/%3E%3C/svg%3E');--md-admonition-icon--bug:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%2016%2016%22%3E%3Cpath%20d%3D%22M4.72.22a.75.75%200%200%201%201.06%200l1%20.999a3.5%203.5%200%200%201%202.441%200l.999-1a.748.748%200%200%201%201.265.332.75.75%200%200%201-.205.729l-.775.776c.616.63.995%201.493.995%202.444v.327q0%20.15-.025.292c.408.14.764.392%201.029.722l1.968-.787a.75.75%200%200%201%20.556%201.392L13%207.258V9h2.25a.75.75%200%200%201%200%201.5H13v.5q-.002.615-.141%201.186l2.17.868a.75.75%200%200%201-.557%201.392l-2.184-.873A5%205%200%200%201%208%2016a5%205%200%200%201-4.288-2.427l-2.183.873a.75.75%200%200%201-.558-1.392l2.17-.868A5%205%200%200%201%203%2011v-.5H.75a.75.75%200%200%201%200-1.5H3V7.258L.971%206.446a.75.75%200%200%201%20.558-1.392l1.967.787c.265-.33.62-.583%201.03-.722a1.7%201.7%200%200%201-.026-.292V4.5c0-.951.38-1.814.995-2.444L4.72%201.28a.75.75%200%200%201%200-1.06m.53%206.28a.75.75%200%200%200-.75.75V11a3.5%203.5%200%201%200%207%200V7.25a.75.75%200%200%200-.75-.75ZM6.173%205h3.654A.17.17%200%200%200%2010%204.827V4.5a2%202%200%201%200-4%200v.327c0%20.096.077.173.173.173%22/%3E%3C/svg%3E');--md-admonition-icon--example:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%2016%2016%22%3E%3Cpath%20d%3D%22M5%205.782V2.5h-.25a.75.75%200%200%201%200-1.5h6.5a.75.75%200%200%201%200%201.5H11v3.282l3.666%205.76C15.619%2013.04%2014.543%2015%2012.767%2015H3.233c-1.776%200-2.852-1.96-1.899-3.458Zm-2.4%206.565a.75.75%200%200%200%20.633%201.153h9.534a.75.75%200%200%200%20.633-1.153L12.225%2010.5h-8.45ZM9.5%202.5h-3V6c0%20.143-.04.283-.117.403L4.73%209h6.54L9.617%206.403A.75.75%200%200%201%209.5%206Z%22/%3E%3C/svg%3E');--md-admonition-icon--quote:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%2016%2016%22%3E%3Cpath%20d%3D%22M1.75%202.5h10.5a.75.75%200%200%201%200%201.5H1.75a.75.75%200%200%201%200-1.5m4%205h8.5a.75.75%200%200%201%200%201.5h-8.5a.75.75%200%200%201%200-1.5m0%205h8.5a.75.75%200%200%201%200%201.5h-8.5a.75.75%200%200%201%200-1.5M2.5%207.75v6a.75.75%200%200%201-1.5%200v-6a.75.75%200%200%201%201.5%200%22/%3E%3C/svg%3E');}</style>



    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CFira+Code:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Fira Code"}</style>
      
    
    
      <link rel="stylesheet" href="../../_core/stylesheets/sg_colors.css">
    
      <link rel="stylesheet" href="../../_core/stylesheets/extra.css">
    
      <link rel="stylesheet" href="../../_core/stylesheets/navbar.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="slate" data-md-color-primary="custom" data-md-color-accent="custom">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#g4x-output-structure" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Singular Genomics Docs" class="md-header__button md-logo" aria-label="Singular Genomics Docs" data-md-component="logo">
      
  <img src="../../_core/assets/sg_logo_text.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Singular Genomics Docs
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              G4X-output
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="custom" data-md-color-accent="custom"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6m0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4M7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="custom" data-md-color-accent="custom"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91s2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/Singular-Genomics/singular-genomics.github.io" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    singular-genomics.github.io
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Singular Genomics Docs" class="md-nav__button md-logo" aria-label="Singular Genomics Docs" data-md-component="logo">
      
  <img src="../../_core/assets/sg_logo_text.svg" alt="logo">

    </a>
    Singular Genomics Docs
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/Singular-Genomics/singular-genomics.github.io" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    singular-genomics.github.io
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Docs Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
      
        
          
        
      
        
      
        
      
    
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    G4X-data
    
  </span>
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_2" id="__nav_2_label" tabindex="">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            G4X-data
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    G4X-output
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    G4X-output
    
  </span>
  

      </a>
      
        

  

<nav class="md-nav md-nav--secondary" aria-label="On this page">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      On this page
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
    <a href="#sample-directory-tree" class="md-nav__link">
      <span class="md-ellipsis">
        Sample directory tree
      </span>
    </a>

    <!-- Table of contents list -->
    
  </li>
      
        <li class="md-nav__item">
    <a href="#sample-sub-directory-reference" class="md-nav__link">
      <span class="md-ellipsis">
        Sample sub-directory reference
      </span>
    </a>

    <!-- Table of contents list -->
    
      <nav class="md-nav" aria-label="Sample sub-directory reference">
        <ul class="md-nav__list">
          
          
            <li class="md-nav__item">
    <a href="#root-of-sample_folder" class="md-nav__link">
      <span class="md-ellipsis">
        root of sample_folder
      </span>
    </a>

    <!-- Table of contents list -->
    
  </li>
          
          
          
            <li class="md-nav__item">
    <a href="#diagnostics" class="md-nav__link">
      <span class="md-ellipsis">
        diagnostics/
      </span>
    </a>

    <!-- Table of contents list -->
    
  </li>
          
          
          
            <li class="md-nav__item">
    <a href="#g4x_viewer" class="md-nav__link">
      <span class="md-ellipsis">
        g4x_viewer/
      </span>
    </a>

    <!-- Table of contents list -->
    
  </li>
          
          
          
            <li class="md-nav__item">
    <a href="#metrics" class="md-nav__link">
      <span class="md-ellipsis">
        metrics/
      </span>
    </a>

    <!-- Table of contents list -->
    
  </li>
          
          
          
            <li class="md-nav__item">
    <a href="#h_and_e" class="md-nav__link">
      <span class="md-ellipsis">
        h_and_e/
      </span>
    </a>

    <!-- Table of contents list -->
    
  </li>
          
          
          
            <li class="md-nav__item">
    <a href="#protein-protein-runs-only" class="md-nav__link">
      <span class="md-ellipsis">
        protein/ (protein runs only)
      </span>
    </a>

    <!-- Table of contents list -->
    
  </li>
          
          
          
            <li class="md-nav__item">
    <a href="#rna" class="md-nav__link">
      <span class="md-ellipsis">
        rna/
      </span>
    </a>

    <!-- Table of contents list -->
    
  </li>
          
          
          
            <li class="md-nav__item">
    <a href="#segmentation" class="md-nav__link">
      <span class="md-ellipsis">
        segmentation/
      </span>
    </a>

    <!-- Table of contents list -->
    
  </li>
          
          
          
            <li class="md-nav__item">
    <a href="#single_cell_data" class="md-nav__link">
      <span class="md-ellipsis">
        single_cell_data/
      </span>
    </a>

    <!-- Table of contents list -->
    
  </li>
          
          
        </ul>
      </nav>
    
  </li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../data_import/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    data import
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
    
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../g4x_tutorials/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    G4X-tutorials
    
  </span>
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_3" id="__nav_3_label" tabindex="">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            G4X-tutorials
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../g4x_tutorials/plot_spatial/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    plot spatial
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../g4x_tutorials/segment_data/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    segment data
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../changelog/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    changelog
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  

<nav class="md-nav md-nav--secondary" aria-label="On this page">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      On this page
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
    <a href="#sample-directory-tree" class="md-nav__link">
      <span class="md-ellipsis">
        Sample directory tree
      </span>
    </a>

    <!-- Table of contents list -->
    
  </li>
      
        <li class="md-nav__item">
    <a href="#sample-sub-directory-reference" class="md-nav__link">
      <span class="md-ellipsis">
        Sample sub-directory reference
      </span>
    </a>

    <!-- Table of contents list -->
    
      <nav class="md-nav" aria-label="Sample sub-directory reference">
        <ul class="md-nav__list">
          
          
            <li class="md-nav__item">
    <a href="#root-of-sample_folder" class="md-nav__link">
      <span class="md-ellipsis">
        root of sample_folder
      </span>
    </a>

    <!-- Table of contents list -->
    
  </li>
          
          
          
            <li class="md-nav__item">
    <a href="#diagnostics" class="md-nav__link">
      <span class="md-ellipsis">
        diagnostics/
      </span>
    </a>

    <!-- Table of contents list -->
    
  </li>
          
          
          
            <li class="md-nav__item">
    <a href="#g4x_viewer" class="md-nav__link">
      <span class="md-ellipsis">
        g4x_viewer/
      </span>
    </a>

    <!-- Table of contents list -->
    
  </li>
          
          
          
            <li class="md-nav__item">
    <a href="#metrics" class="md-nav__link">
      <span class="md-ellipsis">
        metrics/
      </span>
    </a>

    <!-- Table of contents list -->
    
  </li>
          
          
          
            <li class="md-nav__item">
    <a href="#h_and_e" class="md-nav__link">
      <span class="md-ellipsis">
        h_and_e/
      </span>
    </a>

    <!-- Table of contents list -->
    
  </li>
          
          
          
            <li class="md-nav__item">
    <a href="#protein-protein-runs-only" class="md-nav__link">
      <span class="md-ellipsis">
        protein/ (protein runs only)
      </span>
    </a>

    <!-- Table of contents list -->
    
  </li>
          
          
          
            <li class="md-nav__item">
    <a href="#rna" class="md-nav__link">
      <span class="md-ellipsis">
        rna/
      </span>
    </a>

    <!-- Table of contents list -->
    
  </li>
          
          
          
            <li class="md-nav__item">
    <a href="#segmentation" class="md-nav__link">
      <span class="md-ellipsis">
        segmentation/
      </span>
    </a>

    <!-- Table of contents list -->
    
  </li>
          
          
          
            <li class="md-nav__item">
    <a href="#single_cell_data" class="md-nav__link">
      <span class="md-ellipsis">
        single_cell_data/
      </span>
    </a>

    <!-- Table of contents list -->
    
  </li>
          
          
        </ul>
      </nav>
    
  </li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<p><br></p>
<h1 id="g4x-output-structure"><a class="toclink" href="#g4x-output-structure">G4X-output structure</a></h1>
<h2 id="sample-directory-tree"><a class="toclink" href="#sample-directory-tree">Sample directory tree</a></h2>
<hr />
<p>Directory structure depends on run type.</p>
<div class="tabbed-set tabbed-alternate" data-tabs="1:2"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><div class="tabbed-labels"><label for="__tabbed_1_1">Transcript &amp; Protein</label><label for="__tabbed_1_2">Transcript</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="language-text highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>&lt;sample_root&gt;
<a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>│
<a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a>├── diagnostics
<a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a>│   └── transcript_table.parquet
<a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a>│
<a id="__codelineno-0-6" name="__codelineno-0-6" href="#__codelineno-0-6"></a>├── g4x_viewer 
<a id="__codelineno-0-7" name="__codelineno-0-7" href="#__codelineno-0-7"></a>│   ├── &lt;sample_1&gt;.bin
<a id="__codelineno-0-8" name="__codelineno-0-8" href="#__codelineno-0-8"></a>│   ├── &lt;sample_1&gt;.ome.tiff              
<a id="__codelineno-0-9" name="__codelineno-0-9" href="#__codelineno-0-9"></a>│   ├── &lt;sample_1&gt;.tar
<a id="__codelineno-0-10" name="__codelineno-0-10" href="#__codelineno-0-10"></a>│   ├── &lt;sample_1&gt;_HE.ome.tiff
<a id="__codelineno-0-11" name="__codelineno-0-11" href="#__codelineno-0-11"></a>│   ├── &lt;sample_1&gt;_nuclear.ome.tiff
<a id="__codelineno-0-12" name="__codelineno-0-12" href="#__codelineno-0-12"></a>│   └── &lt;sample_1&gt;_run_metadata.json
<a id="__codelineno-0-13" name="__codelineno-0-13" href="#__codelineno-0-13"></a>│
<a id="__codelineno-0-14" name="__codelineno-0-14" href="#__codelineno-0-14"></a>├── h_and_e 
<a id="__codelineno-0-15" name="__codelineno-0-15" href="#__codelineno-0-15"></a>│   ├── eosin.jp2
<a id="__codelineno-0-16" name="__codelineno-0-16" href="#__codelineno-0-16"></a>│   ├── eosin_thumbnail.png
<a id="__codelineno-0-17" name="__codelineno-0-17" href="#__codelineno-0-17"></a>│   ├── h_and_e.jp2
<a id="__codelineno-0-18" name="__codelineno-0-18" href="#__codelineno-0-18"></a>│   ├── h_and_e_thumbnail.jpg
<a id="__codelineno-0-19" name="__codelineno-0-19" href="#__codelineno-0-19"></a>│   ├── nuclear.jp2
<a id="__codelineno-0-20" name="__codelineno-0-20" href="#__codelineno-0-20"></a>│   └── nuclear_thumbnail.png
<a id="__codelineno-0-21" name="__codelineno-0-21" href="#__codelineno-0-21"></a>│
<a id="__codelineno-0-22" name="__codelineno-0-22" href="#__codelineno-0-22"></a>├── metrics 
<a id="__codelineno-0-23" name="__codelineno-0-23" href="#__codelineno-0-23"></a>│   ├── core_metrics.csv
<a id="__codelineno-0-24" name="__codelineno-0-24" href="#__codelineno-0-24"></a>|   ├── protein_core_metrics.csv
<a id="__codelineno-0-25" name="__codelineno-0-25" href="#__codelineno-0-25"></a>│   └── per_area_metrics.csv
<a id="__codelineno-0-26" name="__codelineno-0-26" href="#__codelineno-0-26"></a>│
<a id="__codelineno-0-27" name="__codelineno-0-27" href="#__codelineno-0-27"></a>├── protein                             
<a id="__codelineno-0-28" name="__codelineno-0-28" href="#__codelineno-0-28"></a>│   ├── &lt;protein_1&gt;.jp2
<a id="__codelineno-0-29" name="__codelineno-0-29" href="#__codelineno-0-29"></a>│   ├── &lt;protein_1&gt;_thumbnail.png
<a id="__codelineno-0-30" name="__codelineno-0-30" href="#__codelineno-0-30"></a>│   ├── &lt;protein_2&gt;.jp2
<a id="__codelineno-0-31" name="__codelineno-0-31" href="#__codelineno-0-31"></a>│   ├── &lt;protein_2&gt;_thumbnail.png
<a id="__codelineno-0-32" name="__codelineno-0-32" href="#__codelineno-0-32"></a>│   └── …
<a id="__codelineno-0-33" name="__codelineno-0-33" href="#__codelineno-0-33"></a>│
<a id="__codelineno-0-34" name="__codelineno-0-34" href="#__codelineno-0-34"></a>├── protein_panel.csv                   
<a id="__codelineno-0-35" name="__codelineno-0-35" href="#__codelineno-0-35"></a>│
<a id="__codelineno-0-36" name="__codelineno-0-36" href="#__codelineno-0-36"></a>├── rna
<a id="__codelineno-0-37" name="__codelineno-0-37" href="#__codelineno-0-37"></a>│   └── transcript_table.csv.gz
<a id="__codelineno-0-38" name="__codelineno-0-38" href="#__codelineno-0-38"></a>│
<a id="__codelineno-0-39" name="__codelineno-0-39" href="#__codelineno-0-39"></a>├── run_meta.json
<a id="__codelineno-0-40" name="__codelineno-0-40" href="#__codelineno-0-40"></a>│
<a id="__codelineno-0-41" name="__codelineno-0-41" href="#__codelineno-0-41"></a>├── samplesheet.csv
<a id="__codelineno-0-42" name="__codelineno-0-42" href="#__codelineno-0-42"></a>│
<a id="__codelineno-0-43" name="__codelineno-0-43" href="#__codelineno-0-43"></a>├── segmentation
<a id="__codelineno-0-44" name="__codelineno-0-44" href="#__codelineno-0-44"></a>│   └── segmentation_mask.npz
<a id="__codelineno-0-45" name="__codelineno-0-45" href="#__codelineno-0-45"></a>│
<a id="__codelineno-0-46" name="__codelineno-0-46" href="#__codelineno-0-46"></a>├── single_cell_data
<a id="__codelineno-0-47" name="__codelineno-0-47" href="#__codelineno-0-47"></a>│   ├── cell_by_protein.csv.gz          
<a id="__codelineno-0-48" name="__codelineno-0-48" href="#__codelineno-0-48"></a>│   ├── cell_by_transcript.csv.gz
<a id="__codelineno-0-49" name="__codelineno-0-49" href="#__codelineno-0-49"></a>│   ├── cell_metadata.csv.gz
<a id="__codelineno-0-50" name="__codelineno-0-50" href="#__codelineno-0-50"></a>│   ├── clustering_umap.csv.gz
<a id="__codelineno-0-51" name="__codelineno-0-51" href="#__codelineno-0-51"></a>│   ├── dgex.csv.gz
<a id="__codelineno-0-52" name="__codelineno-0-52" href="#__codelineno-0-52"></a>│   └── feature_matrix.h5
<a id="__codelineno-0-53" name="__codelineno-0-53" href="#__codelineno-0-53"></a>│
<a id="__codelineno-0-54" name="__codelineno-0-54" href="#__codelineno-0-54"></a>├── summary_&lt;sample_1&gt;.html
<a id="__codelineno-0-55" name="__codelineno-0-55" href="#__codelineno-0-55"></a>└── transcript_panel.csv
</code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-text highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>&lt;sample_root&gt;
<a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a>│
<a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a>├── diagnostics
<a id="__codelineno-1-4" name="__codelineno-1-4" href="#__codelineno-1-4"></a>│   └── transcript_table.parquet
<a id="__codelineno-1-5" name="__codelineno-1-5" href="#__codelineno-1-5"></a>│
<a id="__codelineno-1-6" name="__codelineno-1-6" href="#__codelineno-1-6"></a>├── g4x_viewer
<a id="__codelineno-1-7" name="__codelineno-1-7" href="#__codelineno-1-7"></a>│   ├── &lt;sample_1&gt;.bin
<a id="__codelineno-1-8" name="__codelineno-1-8" href="#__codelineno-1-8"></a>│   ├── &lt;sample_1&gt;.tar
<a id="__codelineno-1-9" name="__codelineno-1-9" href="#__codelineno-1-9"></a>│   ├── &lt;sample_1&gt;_HE.ome.tiff
<a id="__codelineno-1-10" name="__codelineno-1-10" href="#__codelineno-1-10"></a>│   ├── &lt;sample_1&gt;_nuclear.ome.tiff
<a id="__codelineno-1-11" name="__codelineno-1-11" href="#__codelineno-1-11"></a>│   └── &lt;sample_1&gt;_run_metadata.json
<a id="__codelineno-1-12" name="__codelineno-1-12" href="#__codelineno-1-12"></a>│
<a id="__codelineno-1-13" name="__codelineno-1-13" href="#__codelineno-1-13"></a>├── h_and_e
<a id="__codelineno-1-14" name="__codelineno-1-14" href="#__codelineno-1-14"></a>│   ├── eosin.jp2
<a id="__codelineno-1-15" name="__codelineno-1-15" href="#__codelineno-1-15"></a>│   ├── eosin_thumbnail.png
<a id="__codelineno-1-16" name="__codelineno-1-16" href="#__codelineno-1-16"></a>│   ├── h_and_e.jp2
<a id="__codelineno-1-17" name="__codelineno-1-17" href="#__codelineno-1-17"></a>│   ├── h_and_e_thumbnail.jpg
<a id="__codelineno-1-18" name="__codelineno-1-18" href="#__codelineno-1-18"></a>│   ├── nuclear.jp2
<a id="__codelineno-1-19" name="__codelineno-1-19" href="#__codelineno-1-19"></a>│   └── nuclear_thumbnail.png
<a id="__codelineno-1-20" name="__codelineno-1-20" href="#__codelineno-1-20"></a>│
<a id="__codelineno-1-21" name="__codelineno-1-21" href="#__codelineno-1-21"></a>├── metrics
<a id="__codelineno-1-22" name="__codelineno-1-22" href="#__codelineno-1-22"></a>│   ├── core_metrics.csv
<a id="__codelineno-1-23" name="__codelineno-1-23" href="#__codelineno-1-23"></a>│   └── per_area_metrics.csv
<a id="__codelineno-1-24" name="__codelineno-1-24" href="#__codelineno-1-24"></a>│
<a id="__codelineno-1-25" name="__codelineno-1-25" href="#__codelineno-1-25"></a>├── rna
<a id="__codelineno-1-26" name="__codelineno-1-26" href="#__codelineno-1-26"></a>│   └── transcript_table.csv.gz
<a id="__codelineno-1-27" name="__codelineno-1-27" href="#__codelineno-1-27"></a>│
<a id="__codelineno-1-28" name="__codelineno-1-28" href="#__codelineno-1-28"></a>├── run_meta.json
<a id="__codelineno-1-29" name="__codelineno-1-29" href="#__codelineno-1-29"></a>│
<a id="__codelineno-1-30" name="__codelineno-1-30" href="#__codelineno-1-30"></a>├── samplesheet.csv
<a id="__codelineno-1-31" name="__codelineno-1-31" href="#__codelineno-1-31"></a>│
<a id="__codelineno-1-32" name="__codelineno-1-32" href="#__codelineno-1-32"></a>├── segmentation
<a id="__codelineno-1-33" name="__codelineno-1-33" href="#__codelineno-1-33"></a>│   └── segmentation_mask.npz
<a id="__codelineno-1-34" name="__codelineno-1-34" href="#__codelineno-1-34"></a>│
<a id="__codelineno-1-35" name="__codelineno-1-35" href="#__codelineno-1-35"></a>├── single_cell_data
<a id="__codelineno-1-36" name="__codelineno-1-36" href="#__codelineno-1-36"></a>│   ├── cell_by_transcript.csv.gz
<a id="__codelineno-1-37" name="__codelineno-1-37" href="#__codelineno-1-37"></a>│   ├── cell_metadata.csv.gz
<a id="__codelineno-1-38" name="__codelineno-1-38" href="#__codelineno-1-38"></a>│   ├── clustering_umap.csv.gz
<a id="__codelineno-1-39" name="__codelineno-1-39" href="#__codelineno-1-39"></a>│   ├── dgex.csv.gz
<a id="__codelineno-1-40" name="__codelineno-1-40" href="#__codelineno-1-40"></a>│   └── feature_matrix.h5
<a id="__codelineno-1-41" name="__codelineno-1-41" href="#__codelineno-1-41"></a>│
<a id="__codelineno-1-42" name="__codelineno-1-42" href="#__codelineno-1-42"></a>├── summary_&lt;sample_1&gt;.html
<a id="__codelineno-1-43" name="__codelineno-1-43" href="#__codelineno-1-43"></a>└── transcript_panel.csv
</code></pre></div>
</div>
</div>
</div>
<p><br></p>
<h2 id="sample-sub-directory-reference"><a class="toclink" href="#sample-sub-directory-reference">Sample sub-directory reference</a></h2>
<hr />
<h3 id="root-of-sample_folder"><a class="toclink" href="#root-of-sample_folder">root of sample_folder</a></h3>
<blockquote>
<p><code>run_meta.json:</code> JSON file containing versioning information for the panels used and analysis pipelines as well as the ID for the sequencer on which the experiment was run.</p>
<p><code>samplesheet.csv:</code> CSV file containing detailed run information. Details the experimental design, flow cell layout, tissue type, panel utilized, etc. This file is useful for analysis, as it designates where each tissue section is positioned on the flow cell.</p>
<p><code>summary_&lt;sample_id&gt;.html:</code> HTML file which gives a high level overview of the experiment outputs, data quality, and performance for the selected tissue block.</p>
<p><code>transcript_panel.csv:</code> CSV file containing a full list of all targeted genes in this experiment and the panel(s) which they originated from.</p>
<p><code>protein_panel.csv:</code> CSV file containing a full list of all targeted proteins in this experiment and the panel(s) which they originated from. <em>Multiomics runs only</em></p>
</blockquote>
<p><br></p>
<h3 id="diagnostics"><a class="toclink" href="#diagnostics">diagnostics/</a></h3>
<blockquote>
<p><code>transcript_table.parquet:</code> Parquet file containing all decoded and non-decoded transcripts and associated metadata (e.g. spatial coordinate, gene identity, cell identity (if assigned to a cell), quality score, sequence). Parquet files can be loaded in Python using the polars, fastparquet, pandas, and pyarrow packages.</p>
</blockquote>
<details class="note">
<summary>Expand to see column descriptions</summary>
<table>
<thead>
<tr>
<th>column</th>
<th>type</th>
<th>description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>x_coord_shift</code></td>
<td><em>float</em></td>
<td>The x coordinate for the transcript (shifted to global coordinates)</td>
</tr>
<tr>
<td><code>y_coord_shift</code></td>
<td><em>float</em></td>
<td>The y coordinate for the transcript (shifted to global coordinates)</td>
</tr>
<tr>
<td><code>z</code></td>
<td><em>int</em></td>
<td>z layer of identified transcript</td>
</tr>
<tr>
<td><code>demuxed</code></td>
<td><em>bool</em></td>
<td>Whether or not the transcript was demultiplexed</td>
</tr>
<tr>
<td><code>transcript_condensed</code></td>
<td><em>str</em></td>
<td>Shortened name of transcript</td>
</tr>
<tr>
<td><code>meanQS</code></td>
<td><em>float</em></td>
<td>Mean quality score for the transcript</td>
</tr>
<tr>
<td><code>cell_id</code></td>
<td><em>uint</em></td>
<td>Cell ID</td>
</tr>
<tr>
<td><code>sequence_to_demux</code></td>
<td><em>str</em></td>
<td>Sequence identified that will be demultiplexed</td>
</tr>
<tr>
<td><code>transcript</code></td>
<td><em>str</em></td>
<td>Long form transcript name (specific to single probe)</td>
</tr>
<tr>
<td><code>TXUID</code></td>
<td><em>str</em></td>
<td>Unique identifier for the transcript</td>
</tr>
</tbody>
</table>
</details>
<p><br> </p>
<h3 id="g4x_viewer"><a class="toclink" href="#g4x_viewer">g4x_viewer/</a></h3>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>The G4X-viewer is a web-based tool for visualizing and exploring G4X-data. All files in this directory are designed to be loaded into and explored with the G4X-viewer. For more information on how to use the G4X-viewer, see <a href="https://docs.singulargenomics.com/G4X-viewer/">G4X-viewer</a>.</p>
</div>
<blockquote>
<p><code>&lt;sample_id&gt;.bin:</code> Binary file containing the segmentation mask for the stitched image. Can be easily read in Python with numpy.</p>
<p><code>&lt;sample_id&gt;.ome.tiff:</code> Multidimensional OME-TIFF image file. On windows, this may appear as <code>&lt;sample_id&gt;.ome</code>. This image contains aggregated images for all protein targets as well as nuclear stain. Can be loaded into any standard ome.tiff readers, including our G4X-viewer, and napari. <em>Multiomics runs only.</em>.</p>
<p><code>&lt;sample_id&gt;_HE.ome.tiff:</code> OME-TIFF image file containing the fH&amp;E stain images. Can be loaded into any standard OME-TIFF readers, including our G4X-viewer and napari.</p>
<p><code>&lt;sample_id&gt;_nuclear.ome.tiff:</code> OME-TIFF image file containing the nuclear stain images. Can be loaded into any standard OME-TIFF readers, including our G4X-viewer and napari.</p>
<p><code>&lt;sample_id&gt;_run_metadata.json:</code> JSON file containing much of the same information as the run_meta.json along with extra core metrics information (such as tissue area, total tx, etc).</p>
<p><code>&lt;sample_id&gt;.tar:</code> Tarball containing all other files from this directory bundled into one file. This can be loaded into the G4X-viewer directly with the “single file upload” option to avoid dragging each file individually. May take longer to load than the individual files due to needing to untar the components before displaying on the Viewer.</p>
</blockquote>
<p><br></p>
<h3 id="metrics"><a class="toclink" href="#metrics">metrics/</a></h3>
<blockquote>
<p><code>core_metrics.csv:</code> CSV file containing a set of core metrics for the tissue block including total transcripts, total area, number of cells and more.</p>
<p><code>protein_core_metrics.csv:</code> CSV file containing a set of core protein metrics for the tissue block including SNR, background intensity, and Fisher's exact scores for the co-occurrence of the protein signal with its associated transcript signal (<code>&lt;protein&gt;_fisher_score</code>) and a random background (<code>&lt;protein&gt;_fisher_score_background</code>). These scores indicate the likelihood of the signal being true signal compared to the measured background. <em>Multiomics runs only.</em></p>
<p><code>per_area_metrics.csv:</code> CSV file containing a set of per-area metrics for the tissue block (coordinate location, number of transcripts, and number of cells), separated out into images from before the images were stitched together into one whole block.</p>
</blockquote>
<p><br></p>
<h3 id="h_and_e"><a class="toclink" href="#h_and_e">h_and_e/</a></h3>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>The .jp2 images in this folder and the <code>/protein/</code> folder are suitable to use for both nuclear and cytoplasmic segmentation. For more information on how you might do this, see <a href="../../g4x_tutorials/segment_data/">segment data</a>.</p>
</div>
<blockquote>
<p><code>eosin.jp2:</code> Full-sized eosin stained JPEG image used for analysis purposes for selected tissue block.</p>
<p><code>eosin_thumbnail.png:</code> Downsampled PNG image from the .jp2 file for easier viewing of the eosin stain for selected tissue block.</p>
<p><code>h_and_e.jp2:</code> Full-sized fH&amp;E JPEG image used for analysis purposes for selected tissue block.</p>
<p><code>h_and_e_thumbnail.jpg:</code> Downsampled PNG image from the .jp2 file for easier viewing of the fH&amp;E stain for selected tissue block.</p>
<p><code>nuclear.jp2:</code> Full-sized nuclear stained JPEG image used for analysis purposes for selected tissue block.</p>
<p><code>nuclear_thumbnail.png:</code> Downsampled PNG image from the .jp2 file for easier viewing of the nuclear stain for selected tissue block.</p>
</blockquote>
<p><br></p>
<h3 id="protein-protein-runs-only"><a class="toclink" href="#protein-protein-runs-only">protein/ <em>(protein runs only)</em></a></h3>
<blockquote>
<p><code>&lt;protein_name&gt;.jp2:</code> Full-sized JPEG image used for analysis purposes. Shows the <code>&lt;protein_name&gt;</code> stain for selected tissue block.</p>
<p><code>&lt;protein_name&gt;_thumbnail.png:</code> Downsampled PNG image of the .jp2 file for easier viewing. Shows the <code>&lt;protein_name&gt;</code> stain for selected tissue block.</p>
</blockquote>
<p><br></p>
<h3 id="rna"><a class="toclink" href="#rna">rna/</a></h3>
<blockquote>
<p><code>transcript_table.csv.gz:</code> CSV file containing a transcript table showing all demuxed transcripts on the whole tissue block. Contains coordinate information, z-layer, gene identity, and cell_id fields. All transcripts here are high confidence transcripts post-filtering and processing.</p>
</blockquote>
<p><br></p>
<h3 id="segmentation"><a class="toclink" href="#segmentation">segmentation/</a></h3>
<blockquote>
<p><code>segmentation_mask.npz:</code> Compressed numpy array file containing the segmentation mask. This can be easily read with the <code>numpy.load()</code> function.</p>
</blockquote>
<p><br></p>
<h3 id="single_cell_data"><a class="toclink" href="#single_cell_data">single_cell_data/</a></h3>
<blockquote>
<p><code>cell_by_protein.csv.gz:</code> Gzipped CSV file in a cell x protein intensity format. Each entry in the table is the average protein intensity for a given protein in a given cell. <em>Multiomics runs only.</em></p>
<p><code>cell_by_transcript.csv.gz:</code> Gzipped CSV file in a cell x transcript format. Each entry in the table is the counts for a given transcript in a given cell.</p>
<p><code>cell_metadata.csv.gz:</code> Gzipped CSV file containing the metadata associated with each cell, including cell_id, protein mean intensity, and transcript counts per cell, per transcript species. This is needed to launch a Seurat object and perform downstream analyses. For more information, see <a href="../data_import/">data import</a>.</p>
</blockquote>
<details class="note">
<summary>Expand to see column descriptions</summary>
<table>
<thead>
<tr>
<th>name</th>
<th>type</th>
<th>description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>label</code></td>
<td><em>str</em></td>
<td>Cell ID</td>
</tr>
<tr>
<td><code>&lt;protein&gt;_intensity_mean</code></td>
<td><em>float</em></td>
<td>Fluorescence intensity mean for a given protein</td>
</tr>
<tr>
<td><code>cell_id</code></td>
<td><em>str</em></td>
<td>Cell ID for a given cell</td>
</tr>
<tr>
<td><code>cell_x/y</code></td>
<td><em>float</em></td>
<td>Spatial X/Y coordinate for the nuclear segmentation centroid</td>
</tr>
<tr>
<td><code>expanded_cell_x/y</code></td>
<td><em>float</em></td>
<td>Spatial X/Y coordinate for the expanded nuclear segmentation centroid</td>
</tr>
<tr>
<td><code>log1p_n_genes_by_counts</code></td>
<td><em>float</em></td>
<td>Log number of unique genes detected</td>
</tr>
<tr>
<td><code>log1p_total_counts</code></td>
<td><em>float</em></td>
<td>Log number of total transcripts</td>
</tr>
<tr>
<td><code>n_genes_by_counts</code></td>
<td><em>int</em></td>
<td>Number of unique genes detected</td>
</tr>
<tr>
<td><code>nuclei_area</code></td>
<td><em>int</em></td>
<td>Area of the nuclear segmentation</td>
</tr>
<tr>
<td><code>nuclei_expanded_area</code></td>
<td><em>int</em></td>
<td>Area of the expanded nuclear segmentation</td>
</tr>
<tr>
<td><code>total_counts</code></td>
<td><em>int</em></td>
<td>Total transcript counts</td>
</tr>
</tbody>
</table>
</details>
<blockquote>
<p><code>clustering_umap.csv.gz:</code> Gzipped CSV file containing the matrix of cell cluster annotations and UMAP coordinates for each cell. This is used to visualize the clustering of cells in 2D for a given leiden resolution or UMAP embedding setting.</p>
</blockquote>
<details class="note">
<summary>Expand to see column descriptions</summary>
<table>
<thead>
<tr>
<th>column</th>
<th>type</th>
<th>description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>label</code></td>
<td><em>str</em></td>
<td>Cell ID</td>
</tr>
<tr>
<td><code>leiden_&lt;resolution&gt;</code></td>
<td><em>int</em></td>
<td>Leiden cluster identity for the cell at the specified resolution (0.2-1.0)</td>
</tr>
<tr>
<td><code>X_umap_&lt;min_dist&gt;_&lt;spread&gt;_&lt;axis&gt;</code></td>
<td><em>float</em></td>
<td>UMAP coordinate for the given cell with the given min_dist and spread for a given axis (1 is typically x/UMAP1, 2 is typically y/UMAP2)</td>
</tr>
</tbody>
</table>
</details>
<blockquote>
<p><code>dgex.csv.gz:</code> Gzipped CSV file containing the differential gene expression (DGEx) results for the selected tissue block. Columns are detailed below.</p>
</blockquote>
<details class="note">
<summary>Expand to see column descriptions</summary>
<table>
<thead>
<tr>
<th>column</th>
<th>type</th>
<th>description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>names</code></td>
<td><em>str</em></td>
<td>Gene symbol</td>
</tr>
<tr>
<td><code>scores</code></td>
<td><em>float</em></td>
<td>Z-score from Wilcoxon rank-sum test</td>
</tr>
<tr>
<td><code>logfoldchanges</code></td>
<td><em>float</em></td>
<td>LogFoldChange for the given cluster compared to <em>all</em> other clusters combined</td>
</tr>
<tr>
<td><code>pvals</code></td>
<td><em>float</em></td>
<td>P-value from Wilcoxon rank-sum test</td>
</tr>
<tr>
<td><code>pvals_adj</code></td>
<td><em>float</em></td>
<td>Adjusted P-value</td>
</tr>
<tr>
<td><code>pct_nz_group</code></td>
<td><em>float</em></td>
<td>Percentage of non-zero values in the given cluster.</td>
</tr>
<tr>
<td><code>pct_nz_reference</code></td>
<td><em>float</em></td>
<td>Percentage of non-zero values in all cells outside the given cluster.</td>
</tr>
<tr>
<td><code>group</code></td>
<td><em>int</em></td>
<td>Leiden cluster identity</td>
</tr>
<tr>
<td><code>leiden_res</code></td>
<td><em>str</em></td>
<td>Leiden clustering resolution that this entry is derived from (1 per gene per cluster)</td>
</tr>
</tbody>
</table>
</details>
<blockquote>
<p><code>feature_matrix.h5:</code> H5ad file containing the full cell by gene matrix as well as a wide array of metadata and annotations you might want to use for downstream analysis. This file can be loaded into Python and run through scanpy or a number of other pipelines. See <a href="../data_import/">data import</a>. The annotations in this file are detailed below.</p>
</blockquote>
<details class="note">
<summary>Expand to see annotation layer descriptions</summary>
<table>
<thead>
<tr>
<th>name</th>
<th>layer</th>
<th><em>str</em></th>
<th>dimensions</th>
<th>description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>&lt;protein&gt;_intensity_mean</code></td>
<td><em>float</em></td>
<td>obs</td>
<td>ncell x  1</td>
<td>Fluorescence intensity mean for a given protein per cell</td>
</tr>
<tr>
<td><code>cell_id</code></td>
<td><em>str</em></td>
<td>obs</td>
<td>ncell x 1</td>
<td>Cell ID for a given cell</td>
</tr>
<tr>
<td><code>cell_x/y</code></td>
<td><em>float</em></td>
<td>obs</td>
<td>ncell x 1</td>
<td>Spatial X/Y coordinate for the nuclear segmentation centroid</td>
</tr>
<tr>
<td><code>expanded_cell_x/y</code></td>
<td><em>float</em></td>
<td>obs</td>
<td>ncell x 1</td>
<td>Spatial X/Y coordinate for the expanded nuclear segmentation centroid per cell</td>
</tr>
<tr>
<td><code>log1p_n_genes_by_counts</code></td>
<td><em>float</em></td>
<td>obs</td>
<td>ncell x 1</td>
<td>Log number of unique genes detected per cell</td>
</tr>
<tr>
<td><code>log1p_total_counts</code></td>
<td><em>float</em></td>
<td>obs</td>
<td>ncell x 1</td>
<td>Log number of total transcripts per cell</td>
</tr>
<tr>
<td><code>n_genes_by_counts</code></td>
<td><em>int</em></td>
<td>obs</td>
<td>ncell x 1</td>
<td>Number of unique genes detected per cell</td>
</tr>
<tr>
<td><code>nuclei_area</code></td>
<td><em>float</em></td>
<td>obs</td>
<td>ncell x 1</td>
<td>Area of the nuclear segmentation per cell</td>
</tr>
<tr>
<td><code>nuclei_expanded_area</code></td>
<td><em>float</em></td>
<td>obs</td>
<td>ncell x 1</td>
<td>Area of the expanded nuclear segmentation per cell</td>
</tr>
<tr>
<td><code>total_counts</code></td>
<td><em>int</em></td>
<td>obs</td>
<td>ncell x 1</td>
<td>Total transcript counts per cell</td>
</tr>
<tr>
<td><code>gene_id</code></td>
<td><em>str</em></td>
<td>var</td>
<td>ngenes x 1</td>
<td>Gene symbol</td>
</tr>
<tr>
<td><code>log1p_mean_counts</code></td>
<td><em>float</em></td>
<td>var</td>
<td>ngenes x 1</td>
<td>Log mean transcript counts across all cells</td>
</tr>
<tr>
<td><code>log1p_total_counts</code></td>
<td><em>float</em></td>
<td>var</td>
<td>ngenes x 1</td>
<td>Log total transcript counts across all cells</td>
</tr>
<tr>
<td><code>mean_counts</code></td>
<td><em>float</em></td>
<td>var</td>
<td>ngenes x 1</td>
<td>Mean counts of each transcript across all cells</td>
</tr>
<tr>
<td><code>modality</code></td>
<td><em>str</em></td>
<td>var</td>
<td>ngenes x 1</td>
<td>G4X modality (transcript or protein)</td>
</tr>
<tr>
<td><code>n_cells_by_counts</code></td>
<td><em>int</em></td>
<td>var</td>
<td>ngenes x 1</td>
<td>Number of cells with counts of each transcript</td>
</tr>
<tr>
<td><code>pct_dropout_by_counts</code></td>
<td><em>float</em></td>
<td>var</td>
<td>ngenes x 1</td>
<td>Percentage of zero-count cells for each gene</td>
</tr>
<tr>
<td><code>probe_type</code></td>
<td><em>str</em></td>
<td>var</td>
<td>ngenes x 1</td>
<td>Type of probe: Negative control probe/sequence (NCP/NCS) or transcript targeting (targeting)</td>
</tr>
<tr>
<td><code>total_counts</code></td>
<td><em>int</em></td>
<td>var</td>
<td>ngenes x 1</td>
<td>Total transcript counts per gene</td>
</tr>
</tbody>
</table>
</details>
<p><br></p>
<p>⸻</p>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../" class="md-footer__link md-footer__link--prev" aria-label="Previous: Index">
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                Index
              </div>
            </div>
          </a>
        
        
          
          <a href="../data_import/" class="md-footer__link md-footer__link--next" aria-label="Next: data import">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                data import
              </div>
            </div>
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
    <a href="mailto:care@singulargenomics.com" target="_blank" rel="noopener" title="support" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M61.4 64C27.5 64 0 91.5 0 125.4c0 .9 0 1.7.1 2.6H0v256c0 35.3 28.7 64 64 64h384c35.3 0 64-28.7 64-64V128h-.1c0-.9.1-1.7.1-2.6 0-33.9-27.5-61.4-61.4-61.4zM464 192.3V384c0 8.8-7.2 16-16 16H64c-8.8 0-16-7.2-16-16V192.3l154.8 117.4c31.4 23.9 74.9 23.9 106.4 0zM48 125.4c0-7.4 6-13.4 13.4-13.4h389.2c7.4 0 13.4 6 13.4 13.4 0 4.2-2 8.2-5.3 10.7L280.2 271.5c-14.3 10.8-34.1 10.8-48.4 0L53.3 136.1c-3.3-2.5-5.3-6.5-5.3-10.7"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://x.com/SingularGenomic" target="_blank" rel="noopener" title="x.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M357.2 48h70.6L273.6 224.2 455 464H313L201.7 318.6 74.5 464H3.8l164.9-188.5L-5.2 48h145.6l100.5 132.9zm-24.8 373.8h39.1L119.1 88h-42z"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://www.singulargenomics.com/" target="_blank" rel="noopener" title="www.singulargenomics.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="m55.7 199.7 30.9 30.9c6 6 14.1 9.4 22.6 9.4h21.5c8.5 0 16.6 3.4 22.6 9.4l29.3 29.3c6 6 9.4 14.1 9.4 22.6v37.5c0 8.5 3.4 16.6 9.4 22.6l13.3 13.3c6 6 9.4 14.1 9.4 22.6V416c0 17.7 14.3 32 32 32s32-14.3 32-32v-2.7c0-8.5 3.4-16.6 9.4-22.6l45.3-45.3c6-6 9.4-14.1 9.4-22.6v-34.7c0-17.7-14.3-32-32-32h-82.7c-8.5 0-16.6-3.4-22.6-9.4l-16-16c-4.2-4.2-6.6-10-6.6-16 0-12.5 10.1-22.6 22.6-22.6h34.7c12.5 0 22.6-10.1 22.6-22.6 0-6-2.4-11.8-6.6-16l-19.7-19.7c-3.9-3.8-5.9-8.7-5.9-13.8s2-10 5.7-13.7L263 89c5.8-5.8 9.1-13.7 9.1-21.9 0-7.2-2.4-13.7-6.4-18.9-3.2-.1-6.4-.2-9.6-.2-95.4 0-175.7 64.2-200.3 151.7zM464 256c0-34.6-8.4-67.2-23.4-95.8-6.4.9-12.7 3.9-17.9 9.1l-13.4 13.4c-6 6-9.4 14.1-9.4 22.6V240c0 17.7 14.3 32 32 32H456c2.5 0 5-.3 7.3-.8.4-5 .5-10.1.5-15.2zM0 256a256 256 0 1 1 512 0 256 256 0 1 1-512 0"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
      <div class="md-progress" data-md-component="progress" role="progressbar"></div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.path", "navigation.footer", "navigation.top", "navigation.instant", "navigation.instant.prefetch", "navigation.instant.progress", "navigation.sections", "navigation.indexes", "navigation.tracking", "content.code.annotate", "content.code.copy", "content.code.select", "content.tabs.link", "toc.follow", "search.suggest", "search.share"], "search": "../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
    
  </body>
</html>